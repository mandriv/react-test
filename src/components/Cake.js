import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Item, Dropdown, Responsive } from 'semantic-ui-react';
import EditCakeModal from '../components/EditCakeModal';

/*
  CakeSegment description
*/

export default class Cake extends Component {

  state = { editMode: false }

  render() {
    const { editMode } = this.state;
    const { image, title, desc } = this.props.item;
    return (
      <Item>
        <Item.Image size="small" src={image} />
        <Item.Content>
          <Item.Header className="capitalize">
            {title}
          </Item.Header>
          <Item.Meta className="capitalize">
            {desc}
          </Item.Meta>
          {/* Mobile only */}
          <Responsive
            {...Responsive.onlyMobile}
            as={Item.Extra}
          >
            <Dropdown
              icon="settings"
              floated="right"
              className="top left pointing dropdown"
            >
              <Dropdown.Menu>
                <Dropdown.Item
                  onClick={() => this.setState({ editMode: true })}
                >
                  Edit
                </Dropdown.Item>
                <Dropdown.Item
                  onClick={() => this.props.onRemove(this.props.item)}
                >
                  Delete
                </Dropdown.Item>
              </Dropdown.Menu>
            </Dropdown>
          </Responsive>
        </Item.Content>
        {/* Non mobile */}
        <Responsive
          minWidth={Responsive.onlyTablet.minWidth}
        >
          <Dropdown
            icon="settings"
            floated="right"
            className="top right pointing dropdown"
          >
            <Dropdown.Menu>
              <Dropdown.Item
                onClick={() => this.setState({ editMode: true })}
              >
                Edit
              </Dropdown.Item>
              <Dropdown.Item
                onClick={() => this.props.onRemove(this.props.item)}
              >
                Delete
              </Dropdown.Item>
            </Dropdown.Menu>
          </Dropdown>
        </Responsive>
        <EditCakeModal
          cake={this.props.item}
          open={editMode}
          closeIcon
          closeOnDocumentClick
          onClose={() => this.setState({ editMode: false })}
          onSubmit={newCake => this.props.onEdit(newCake)}
        />
      </Item>
    );
  }

}

Cake.propTypes = {
  item: PropTypes.object.isRequired,
  onRemove: PropTypes.func,
  onEdit: PropTypes.func,
};

Cake.defaultProps = {
  onRemove: () => null,
  onEdit: () => null,
};
