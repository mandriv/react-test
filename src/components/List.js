import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Item } from 'semantic-ui-react';

/*
  List container
*/

export default class List extends Component {

  render() {
    const { items, RowComponent, onRowEdit, onRowRemove } = this.props;
    const listItems = items.map(item => (
      <RowComponent
        key={item.id}
        item={item}
        onEdit={row => onRowEdit(row)}
        onRemove={row => onRowRemove(row)}
      />
    ));
    return (
      <Item.Group divided>
        {listItems}
      </Item.Group>
    );
  }

}

List.propTypes = {
  items: PropTypes.array.isRequired,
  RowComponent: PropTypes.func.isRequired,
  onRowEdit: PropTypes.func,
  onRowRemove: PropTypes.func,
};

List.defaultProps = {
  onRowEdit: () => null,
  onRowRemove: () => null,
};
