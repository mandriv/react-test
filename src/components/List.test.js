import React from 'react';
import { shallow } from 'enzyme';
import List from './List';
import Cake from './Cake';

describe('Component: <List>', () => {
  it('Renders <List>', () => {
    const cakesList = [
      {
        title: 'Testy Cake',
        desc: 'description of test cake',
        image: 'https://picsum.photos/400',
        id: '0',
      },
      {
        title: 'Testarossa Cake',
        desc: 'description of this cake',
        image: 'https://picsum.photos/400',
        id: '1',
      },
    ];
    const listComponent = shallow(<List items={cakesList} RowComponent={Cake} />);
    expect(listComponent).toMatchSnapshot();
  });
});
