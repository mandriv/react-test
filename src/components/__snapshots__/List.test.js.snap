// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <List> Renders <List> 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    RowComponent={[Function]}
    items={
        Array [
            Object {
              "desc": "description of test cake",
              "id": "0",
              "image": "https://picsum.photos/400",
              "title": "Testy Cake",
            },
            Object {
              "desc": "description of this cake",
              "id": "1",
              "image": "https://picsum.photos/400",
              "title": "Testarossa Cake",
            },
          ]
    }
    onRowEdit={[Function]}
    onRowRemove={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Cake
          item={
                    Object {
                              "desc": "description of test cake",
                              "id": "0",
                              "image": "https://picsum.photos/400",
                              "title": "Testy Cake",
                            }
          }
          onEdit={[Function]}
          onRemove={[Function]}
/>,
        <Cake
          item={
                    Object {
                              "desc": "description of this cake",
                              "id": "1",
                              "image": "https://picsum.photos/400",
                              "title": "Testarossa Cake",
                            }
          }
          onEdit={[Function]}
          onRemove={[Function]}
/>,
      ],
      "divided": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "item": Object {
            "desc": "description of test cake",
            "id": "0",
            "image": "https://picsum.photos/400",
            "title": "Testy Cake",
          },
          "onEdit": [Function],
          "onRemove": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "item": Object {
            "desc": "description of this cake",
            "id": "1",
            "image": "https://picsum.photos/400",
            "title": "Testarossa Cake",
          },
          "onEdit": [Function],
          "onRemove": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Cake
            item={
                        Object {
                                    "desc": "description of test cake",
                                    "id": "0",
                                    "image": "https://picsum.photos/400",
                                    "title": "Testy Cake",
                                  }
            }
            onEdit={[Function]}
            onRemove={[Function]}
/>,
          <Cake
            item={
                        Object {
                                    "desc": "description of this cake",
                                    "id": "1",
                                    "image": "https://picsum.photos/400",
                                    "title": "Testarossa Cake",
                                  }
            }
            onEdit={[Function]}
            onRemove={[Function]}
/>,
        ],
        "divided": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "item": Object {
              "desc": "description of test cake",
              "id": "0",
              "image": "https://picsum.photos/400",
              "title": "Testy Cake",
            },
            "onEdit": [Function],
            "onRemove": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "item": Object {
              "desc": "description of this cake",
              "id": "1",
              "image": "https://picsum.photos/400",
              "title": "Testarossa Cake",
            },
            "onEdit": [Function],
            "onRemove": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
